.github/workflows/ci.yml

```diff
@@ -0,0 +1,45 @@
+name: Pull request checks
+
+on: [ pull_request ]
+
+jobs:
+  build-ubuntu:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install dependencies
+      run: sudo ./util/ubuntu-setup
+    - name: make kicker
+      run: make kicker
+    - name: make control
+      run: make control
```

Comment:
If we are doing monorepo, this should also cd to the mtrain folder and attempt to build all of the tests there.

Reply:
Done. `make control` already builds most firmware tests (but I created an alias called `make all-tests` to make that more clear for viewing ci. I added building runs for kicker-tests and docs.

Reply:
That's a bug with make control actually looking at it, it should only build the main bin file

Reply:
Compared to the rest of the targets it should be "make -j$(nproc) control" rather than just "make -j$(nproc)". A similar fix should be applied to kicker as well.

Reply:
yeah I thought that was a bit weird. I fixed it like you suggested (you can confirm it worked by reading the GH actions logs since control only creates control.elf and the other control-tests target only creates .elfs for all the tests). 

Other changes, like fixing this bug in the mtrain/ makefile and deleting outdated tests will be done in the monorepo pr since this pr is really just about setting up ci I think.